monkey/monkey_island/cc/resources/agent_events.py

```diff
@@ -41,19 +46,67 @@ def post(self):
         return {}, HTTPStatus.NO_CONTENT
 
     def get(self):
+        try:
+            type_, success = self.parse_event_filter_args()
+        except Exception as err:
+            return {"error": str(err)}, HTTPStatus.UNPROCESSABLE_ENTITY
 
-        events = self._agent_event_repository.get_events()
+        events = self._get_filtered_events(type_, success)
 
+        try:
+            serialized_events = self._serialize_events(events)
+        except (TypeError, ValueError) as err:
+            return {"error": str(err)}, HTTPStatus.INTERNAL_SERVER_ERROR
+
+        return serialized_events, HTTPStatus.OK
+
+    def parse_event_filter_args(self) -> Tuple[Optional[Type[AbstractAgentEvent]], Optional[bool]]:
```

Comment:
Should this be private? If not, should it have a docstring?

Reply:
Private. Good catch. Fixed.